source/dhtslib/sam/reader.d

```diff
@@ -246,27 +247,28 @@ struct SAMReader
     *   auto reads6 = bamfile.query("{HLA-DRB1*12:17}:1-100");
     *   ```
     */ 
-    auto query(string chrom, long start, long end)
+    auto query(CoordSystem cs)(string chrom, Coordinates!cs coords)
     in (!this.header.isNull)
     {
         auto tid = this.header.targetId(chrom);
-        return query(tid, start, end);
+        return query(tid, coords);
     }
 
     /// ditto
-    auto query(int tid, long start, long end)
+    auto query(CoordSystem cs)(int tid, Coordinates!cs coords)
     in (!this.header.isNull)
     {
-        auto itr = sam_itr_queryi(this.idx, tid, start, end);
+        /// convert to zero-based half-open
+        auto newcoords = coords.to!(CoordSystem.zbho);

```

Comment:
This and all other functions like it throughout the PR: is there a way to short-circuit (or does it do it automatically) if you are already in the correct coordinate system?

Reply:
The way ```coordinates.d``` is written if your coordinates are already in Zbho and you try converting to Zbho it will just return itself. So yes, we automatically do this.

Reply:
Sorry, what I meant was skipping the call to `to` template, perhaps with a static if (since the type is known at compile time )

Reply:
We could but that is basically what ```to``` does. It is a static if as well.

Reply:
Not equivalent -- The difference is that currently, a new value is allocated (`auto newcorrds =`) every time, albeit onto the stack. 

In practicve, it probably doesn't matter much, unless queryi is called in a hot piece of code or inner loop (perhaps unlikely).